### ISSUE TITLE
Alternative implementation of apt-get functions.

### ISSUE THREAD

**Issue by xaosfiftytwo** [BunsenLabs/bunsen-common#4]
Available for review at https://github.com/xaosfiftytwo/bunsen-common in x52 branch.

No need to repeat identical ignore-messages.
Grep pattern creation from array of strings.
Everything concerning apt-get concentrated in one internal function that can not be called directly, but only via 'exposed to the world' functions. Almost a public and private interface in bash.
Easily extendable.

Up to you. Do as you please. :)

**Comment by johnraff** [comment]
Thank you.
It needs some careful testing before implementation though.
(For example, safeUpdate() 's checking for the _--ignore_ string is ignored, and all arguments just passed to _aptGet as-is.)

Since there is no actual change in functionality, can I leave this till after the RC2 release?

**Comment by hhhorb** [comment]
That sounds like a good idea, John.

**Comment by xaosfiftytwo** [comment]
@John,
I understand this is an important link in the chain, that needs to be tested thoroughly.
I have no problem with leaving this as is until after the next RC2 release.

BTW: **safeUpdate does take account of an ignore-string** 
safeUpdate passes all args to _aptGet after eventually changing an empty ignore string.
in _aptGet, before calling the real apt-get, an eventual ignore string is captured in array ignore_strings and tested against apt_error after the call to apt-get.

I have overwritten /usr/lib/bunsen/bunsen-common/bl -include-cfg in my system with the alternative version and tested it a.o. by running bl-welcome. (Thats how I noticed the bl-welcome log got appended to)

**Comment by xaosfiftytwo** [comment]
@John,
OK. There was a typo in _aptGet handling the --ignore argument.
Corrected.
Which shows that you are right again. (about postponing I mean)
And also shows that we should perhaps define 'constant strings' as read-only variables once, tp avoid such typos in the future.

**Comment by johnraff** [comment]
There still seems to be an anomaly with safeUpdate() : as far as I can see the argument processing in the function is done with the _local_ variable _ignore_string_, which is then ignored and the initial arguments passed to _aptGet as they were originally. ( `_aptGet "${@}"` ) _aptGet() then does the same processing, based on being called by safeUpdate.

(BTW the other apt functions are handled slightly differently, passing apt-get's command as _aptGet's first argument, and testing that command against what's expected from ${FUNCNAME[1]}.)

```
safeUpdate() {
    local ignore_string=
    [[ $1 = '--ignore' ]] && {
        ignore_string="$2"
        [[ -z "$ignore_string" ]] && ignore_string='^$'
    }
    _aptGet "${@}"
}
```

**Comment by xaosfiftytwo** [comment]
Thanks for catching that John.
During development, I moved ignore string handling from safeUpdate to _aptGet but forgot to remove the latter.
It was working OK though, i.e. it handled any ignore string pased to it. That is why I did not catch it in my tests.

I have now streamlined upgrade handling more with the other actions.
Also included some improvements (see commit message)

Tested following cases:

`
safeUpdate

safeUpdate --ignore "W: Some index files failed to download. They have been ignored, or old ones used instead."

safeUpdate --ignore

safeUpdate --ignore 'ignore string 1' --ignore 'ignore string 2'

safeUpdate --ignore --ignore 'ignore string2'
`
